app/src/main/java/org/mozilla/fenix/components/metrics/AppStartupTypeTelemetry.kt

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.metrics
+
+import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import androidx.lifecycle.ProcessLifecycleOwner
+import org.mozilla.fenix.components.metrics.Event.AppStartupType.Type
+import org.mozilla.fenix.components.metrics.Event.AppStartupType.Type.COLD
+import org.mozilla.fenix.components.metrics.Event.AppStartupType.Type.WARM
+import org.mozilla.fenix.components.metrics.Event.AppStartupType.Type.HOT
+import org.mozilla.fenix.components.metrics.Event.AppStartupType.Type.COLD_SAVED_INSTANCE
+import org.mozilla.fenix.components.metrics.Event.AppStartupType.Type.WARM_SAVED_INSTANCE
+import org.mozilla.fenix.HomeActivity
+
+/**
+ * Tracks the startup type for application through [Event.AppStartupType]. Consider app to be
+ * "opened" only if application was previously backgrounded or application didn't exist.
+ */
+class AppStartupTypeTelemetry(private val metrics: MetricController) : LifecycleObserver {
+
+    init {
+        ProcessLifecycleOwner.get().lifecycle.addObserver(this)
+    }
+
+    // keeps track of whether we already recorded the metric
+    private var isMetricRecorded = false
+
+    // keeps track of whether the onCreate() was called for FenixApplication.
+    // helps choose which startup type to report during onHomeActivityOnCreate()
```

Comment:
nit: This comment could easily go out-of-date: what if it is no longer used to help choose a startup type or it's used for something else? For variables and methods, you can generally let the code do the talking unless it's difficult to come up with a name that accurately represents what's happening at a high-level. I'd remove this comment

Also note that names should generally be focused on what a value represents, not how it's intended to be used: someone can always add code to use your value in a new way and naming it after it's intended use both limits its re-use and is more likely to go out of date and become misleading. (this is a form of duplication: the name duplicates the code) You're doing the right thing with these variables but I just wanted to clarify because I was on the subject. :)